src/index.ts

```diff
@@ -291,6 +302,29 @@ function noProxyFromEnv(envValue?: string) {
 	});
 }
 
+function noProxyFromConfig(noProxy: string[]) {
+	const value = noProxy.map((item) => item.trim().toLowerCase());
+
+	if (value.includes("*")) {
+		return () => true;
+	}
+
+	const filters = value
```

Comment:
Starting from here the rest of the function is the same as for the env variable above. Maybe extract a new function with the common code to avoid duplicating it.

Reply:
I have created a new function `shouldBypassProxy` which is called now called in both `noProxyFromEnv` and `noProxyFromConfig`, thus reducing code duplication